### PR TITLE
Add be.telenet.user.QSSTV

### PR DESCRIPTION
**Pull request by lu9dce**
QSSTV is a program for receiving and transmitting SSTV and HAMDRM
(sometimes called DSSTV). It is compatible with most of MMSSTV and EasyPal

transmitdrmb

### CODE REVIEW

**Comment by lu9dce** [comment]
Well one of the great radio amateurs
this time I decided to use simple to build the flatpak
since many of the programs force to install in / usr
and require a configuration by hand
maybe there another better way .. but I feel more comfortable like this
I also learned not to create the icon and the .desktop if it is in the package
also to make it simple helps me understand the way they are processed
I am trying to add variables to make it more beautiful
being able to build qsstv will help me do the wsjtx
which requires more parameters in the configuration
If wsjtx can be passed to flatpak, I will be able to do any other program more easily and do it in a faster way.
qsstv it took me two days to make it work
... there is always something to learn .. like any good linux user - uneasy mind

**Comment by lu9dce** [comment]
before giving the bot build
I hope they check to see if everything is ok

File: be.telenet.user.qsstv.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2020 hamradio lu9dce -->
+<component type="desktop-application">
+ <id>be.telenet.user.qsstv</id>
+ <metadata_license>CC0-1.0</metadata_license>
+ <project_license>GPL-3.0</project_license>
+ <name>qsstv</name>
```
Comment (AsciiWolf) [inline be.telenet.user.qsstv.metainfo.xml:7]: QSSTV

File: be.telenet.user.qsstv.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ <summary>receiving slow-scan television and fax</summary>
```
Comment (AsciiWolf) [inline be.telenet.user.qsstv.metainfo.xml:8]: **R**eceiving

(uppercase)

Comment (AsciiWolf) [reply be.telenet.user.qsstv.metainfo.xml:8]: Or even better:
Program for receiving and transmitting SSTV and HAMDRM

File: be.telenet.user.qsstv.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ <developer_name>Johan Maes ON4QZ</developer_name>
+ <url type="homepage">http://users.telenet.be/on4qz/qsstv/</url>
+ <description>
+  <p>
+   QSSTV is a program for receiving slow-scan television and fax.
```
Comment (AsciiWolf) [inline be.telenet.user.qsstv.metainfo.xml:13]: From Fedora:
QSSTV is a program for receiving and transmitting SSTV and HAMDRM (sometimes called DSSTV). It is compatible with most of MMSSTV and EasyPal.

**Comment by AsciiWolf** [comment]
bot, build be.telenet.user.qsstv

**Comment by AsciiWolf** [comment]
The provided icon does not seem to be 128x128.

**Comment by lu9dce** [comment]
> The provided icon does not seem to be 128x128.

 - cp $FLATPAK_BUILDER_BUILDDIR/qsstv/icons/qsstv.png $FLATPAK_DEST/share/icons/hicolor/128x128/apps/qsstv.png

**Comment by AsciiWolf** [comment]
Yes. The provided icon seems to be 42x42 which is a small and non-standard, small.

**Comment by AsciiWolf** [comment]
Yes. The provided icon seems to be 42x42 which is a small and non-standard size.

**Comment by AsciiWolf** [comment]
You can probably change the icon size manually to a standard one, for example 36x36 and bundle it with this Flatpak. Then create a `flathub.json` file containing:
```
{
    "skip-icons-check": true
}
```

It will look bad, but at least it will work.

**Comment by lu9dce** [comment]
> {
>     "skip-icons-check": true
> }
> ```

ok

File: be.telenet.user.qsstv.metainfo.xml (hunk 2 above)
Comment (lu9dce) [reply be.telenet.user.qsstv.metainfo.xml:8]: ok

**Comment by lu9dce** [comment]
bot, build be.telenet.user.qsstv

**Comment by lu9dce** [comment]
@AsciiWolf 
I am having trouble doing wsjtx / wsjtx / jtdx / mshv
what are the most used
wsjtz the source comes in 7z mmm and does not recognize the format
jtdx directly throw errors to compile
wsjtz requires an older version of qt
wsjx almost made it
I'm going to keep trying with wsjtx, which is the original and goes with the last qt5
I'm going to make the manifesto that let's say is almost 90% walking
I'll upload it to flatpak to see what I'm getting wrong
but before doing this I will continue uploading other soft that are easier

**Comment by AsciiWolf** [comment]
bot, build be.telenet.user.qsstv

File: be.telenet.user.qsstv.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+command: qsstv
+rename-desktop-file: qsstv.desktop
+rename-icon: qsstv
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=all
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=home
```
Comment (AsciiWolf) [inline be.telenet.user.qsstv.yml:14]: I have tried replacing `--filesystem=home` with `--persist=qsstv`, however the `qsstv` directory was not created inside the sandbox for some reason and also there were some other issues like saved images/templates being created with a `.xdp- prefix` for some reason (maybe a portal bug?).

So, `--filesystem=home` is good for now.

**Comment by AsciiWolf** [comment]
I have tried receive, transmit and other basic functions. Everything seems to work fine. :-)

File: be.telenet.user.qsstv.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2020 hamradio lu9dce -->
+<component type="desktop-application">
+ <id>be.telenet.user.qsstv</id>
+ <metadata_license>CC0-1.0</metadata_license>
+ <project_license>GPL-3.0</project_license>
+ <name>qsstv</name>
+ <summary>QSSTV is a program for receiving and transmitting SSTV and HAMDRM (sometimes called DSSTV). It is compatible with most of MMSSTV and EasyPal.</summary>
```
Comment (AsciiWolf) [inline be.telenet.user.qsstv.metainfo.xml:8]: The summary is too long. Change it to:
QSSTV is a program for receiving and transmitting SSTV and HAMDRM

**Comment by AsciiWolf** [comment]
Please, fix the [name](https://github.com/flathub/flathub/pull/1973#discussion_r535451506) and [summary](https://github.com/flathub/flathub/pull/1973#discussion_r536068885) in AppStream metadata. Otherwise LGTM!

File: be.telenet.user.qsstv.yml (hunk 4 above)
Comment (lu9dce) [reply be.telenet.user.qsstv.yml:14]: qsl

**Comment by lu9dce** [comment]
bot, build be.telenet.user.qsstv

**Comment by lu9dce** [comment]
bot, build be.telenet.user.QSSTV

**Comment by nedrichards** [comment]
bot, build be.telenet.user.QSSTV

**Comment by lu9dce** [comment]
GPL (GNU Public License or General Public License): Basically it defends the access and modification of the code without restrictions, the freedom of distribution and the maintenance of this freedom

flatpak does not comply with this I withdraw the soft

**Comment by barthalion** [comment]
@AsciiWolf Is it something you want to take over?

**Comment by AsciiWolf** [comment]
@barthalion Sorry, but probably nope. At least not now. I already maintain too much stuff and have too little time. :-( I would love to have apps like QSSTV or QLog on Flathub, but on the other hand I don't want to take care of additional stuff myself.

By the way, what should be done with Fldigi and Flrig? @lu9dce Do you want them to be removed from Flathub as well?

**Pull request closed without merging**